kpconv_torch/utils/tester.py

```diff
@@ -347,7 +347,8 @@ def cloud_segmentation_test(self, net, test_loader, config, num_votes=100, debug
                     print(s + "\n")
 
                 # Save real IoU once in a while
-                if int(np.ceil(new_min)) % config.potential_increment == 0:
+                if last_saved_min + config.potential_increment < new_min:
```

Comment:
I did not understand the change of the form of the test. Could you explain it?

Reply:
Old behavior: write prediction results each time the (rounded) min potential is divisible by 10.

Why not, but this is a problem for the functional test purpose: in order to keep the inference functional test short, I prefered to run only a few iterations, then the min potential is smaller than 10. :-)

I first parametrized the "10" value through a new function parameter (available in the CLI, by the way), so as to pass the accurate value for this parameter in the functional test.

But there was still a problem: the min potential does not seem to increase similarly on my laptop and on the different CI jobs (which are run on different Python versions). Let's say the argument was "6": I got [6.74](https://github.com/delhomer/kpconv-torch/actions/runs/7971771163/job/21762155346#step:7:590) (then 7, it is rounded to the upper integer) with Python 3.10 (fail), [6.79](https://github.com/delhomer/kpconv-torch/actions/runs/7971771163/job/21762155125#step:7:599) for Python 3.9 (fail) and [5.79](https://github.com/delhomer/kpconv-torch/actions/runs/7971771163/job/21762154846#step:7:652) for Python 3.8 (OK).

That's why I chose to modify the prediction save rule: they are now saved when min potential increases of a specific increment (which is parameterized). If I saved the predictions when min potential was 15, and if my parameter is 6, I will save predictions again when min potential will reach 21.

Reply:
ok, very clear now!